services/app-api/postgres/submissionWithRevisionsHelpers.ts

```diff
@@ -0,0 +1,34 @@
+import { StateSubmission, StateSubmissionRevision } from '@prisma/client'
+import { StoreError } from './storeError'
+
+export type StateSubmissionWithRevisions = StateSubmission & {
+    revisions: StateSubmissionRevision[]
+}
+
+// Return first revision associated with a submission or return a StoreError if there is submission or revisions
+// used validate prisma results have useable submission
+const getCurrentRevision = (
+    submissionID: string,
+    submissionResult: StateSubmissionWithRevisions | null
+): StateSubmissionRevision | StoreError => {
+    if (!submissionResult)
+        return {
+            code: 'UNEXPECTED_EXCEPTION' as const,
```

Comment:
Ha, this is kinda funny to me, like I’d expect checking that you got a StateSubmission happen before calling getCurrentRevision, but I think this is fine.